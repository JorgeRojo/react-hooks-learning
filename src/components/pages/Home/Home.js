import React from 'react';

function Home() {
  return (
    <div className="container text-center mt-4">
      <h1 className="display-4 mb-4">Home</h1>
      <p>Testing react hooks</p>
    </div>
  );
}

export default Home;
