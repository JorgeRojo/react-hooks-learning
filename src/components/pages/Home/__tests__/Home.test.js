import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';

import Home from '../Home';

describe('Home Component', () => {
  const component = <Home />;

  afterEach(() => {
    cleanup();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
