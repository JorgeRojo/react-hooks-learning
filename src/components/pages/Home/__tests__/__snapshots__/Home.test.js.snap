// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders the component 1`] = `
<div>
  <div
    class="container text-center mt-4"
  >
    <h1
      class="display-4 mb-4"
    >
      Home
    </h1>
    <p>
      Testing react hooks
    </p>
  </div>
</div>
`;
